shell/platform/darwin/macos/framework/Source/FlutterViewController.mm

```diff
@@ -278,10 +365,17 @@ - (void)loadView {
     }
     flutterView = [[FlutterView alloc] initWithMainContext:mainContext reshapeListener:self];
   }
-  self.view = flutterView;
+  FlutterNativeTextFieldHidder* nativeTextFieldHidder =
+      [[FlutterNativeTextFieldHidder alloc] initWithFlutterView:flutterView];
+  [nativeTextFieldHidder addSubview:flutterView];
+  self.view = nativeTextFieldHidder;
+  self.flutterView = flutterView;
 }
 
 - (void)viewDidLoad {
+  FlutterAppDelegate* appDelegate = (FlutterAppDelegate*)[NSApp delegate];
+  self.windowDelegate = [[FlutterWindowDelegate alloc] initWithController:self];
+  appDelegate.mainFlutterWindow.delegate = _windowDelegate;
```

Comment:
This is a non-starter; just as we don't own the application, we don't own the window. This could stomp a developer's window delegate, or they could stomp this.

If a window-level hook is the only way to make this workaround function, then you'll need to wire up an explicit way for an application to delegate to Flutter (likely via FVC) and then make the template set up a skeleton window delegate that explicitly does that forwarding, with a comment explaining why, and what to do it they want to have a different delegate.

Reply:
Thanks for pointing this out. There may be a way to inject field editor without windows delegate if i can create the custom  NSTextFieldCell. I have to test it out.